src/mds/mdstypes.h

```diff
@@ -30,6 +30,7 @@
 #include <boost/pool/pool.hpp>
 #include "include/ceph_assert.h"
 #include <boost/serialization/strong_typedef.hpp>
+#include "common/ceph_json.h"
```

Comment:
same here

Reply:
because 
I need declare callback template functions in mdstypes.h
```
865template<template<typename> class Allocator>
 866-void inode_t<Allocator>::client_ranges_cb(typename inode_t<Allocator>::client_range_map& c, JSONObj *obj){                                                                                                 
 867    int64_t client;                                                                                                                                                                                         
 868    JSONDecoder::decode_json("client", client, obj, true);                                                                                                                                                  
 869    client_writeable_range_t client_range_tmp;                                                                                                                                                              
 870    JSONDecoder::decode_json("byte range", client_range_tmp.range, obj, true);                                                                                                                              
 871    JSONDecoder::decode_json("follows", client_range_tmp.follows.val, obj, true);                                                                                                                           
 872    c[client] = client_range_tmp;                                                                                                                                                                           
 873}
```